server/src/main/java/io/crate/auth/Credentials.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.auth;
+
+import org.elasticsearch.common.settings.SecureString;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Holder for all authentication methods.
+ * username is used for password, cert and trust methods and null otherwise.
+ * password is used for password method and null otherwise.
+ * jwtToken is used for jwt method and null otherwise.
+ */
+public class Credentials {
```

Comment:
Please change to a `record`.

Reply:
It was a record in the first iteration but had to change it - It's mutable because password can be injected in the middle of workflow in the Postgres